.github/workflows/rust-build.yml

```diff
@@ -21,7 +20,11 @@ jobs:
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
+          key:
+            ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}-${{
+            github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
```

Comment:
shouldn't we start from the `${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}-${{ github.run_id }}` until `${{ runner.os }}-cargo`?